src/System.IO.Packaging/tests/Tests.cs

```diff
@@ -3756,6 +3756,22 @@ public void VeryLargePart()
             }
         }
 
+        [Fact]
+        public void SupportNonPartEnumerableDerivedPackageTypes()
+        {
+            // Package is an abstract class that others can derive from. Those derived classes can override GetPartsCore and potentially not
+            // return anything. Furthermore, it is not guaranteed that GetPartsCore will have been called if the package wasn't created using
+            // the Package.Open API, which only ensures that ZipPackage's internal data structures are filled in.
+            var mockPackage = new NonEnumerablePackage();
+
+            var partUri = new Uri("/idontexist.xml", UriKind.Relative);
+            var part = mockPackage.GetPart(partUri);
+
+            Assert.NotNull(part);
+            Assert.Equal(part.Uri, partUri);
+            Assert.IsType(typeof(MockPackagePart), part);
+        }
```

Comment:
Call GetPart again and make sure you get back the same object (reference equals).